geth/node/manager.go

```diff
@@ -545,13 +529,53 @@ func (m *NodeManager) RPCClient() (*rpc.Client, error) {
 		}
 	}
 
+	return m.rpcClient, nil
+}
+
+// RPCUpstreamClient exposes reference to RPC client connected to the running node.
+func (m *NodeManager) RPCUpstreamClient() (*rpc.Client, error) {
+	if m == nil {
+		return nil, ErrInvalidNodeManager
+	}
```

Comment:
Guys, I think it's about time to stop mixing clear business logic with a lot of defensive mechanisms. These errors should be verified in tests.